Develop

frezworx opened the pull request.

frezworx committed:
Update .gitignore to fix and extend ignored entries

Corrected the .env entry and added .venv to the ignore list. This ensures better exclusion of environment files and directories.

frezworx committed:
Add new models: Actor, Genre, CinemaHall, and MovieHall

Introduced models to represent actors, genres, cinema halls, and movies with their attributes and relationships. These changes lay the foundation for managing cinemas, movie metadata, and associations within the application.

frezworx committed:
Add serializers and update models for actors and genres

Introduced new serializers for actors, genres, and cinema halls to handle data serialization. Updated the Movie model to include ManyToMany relationships with Actor and Genre, improving data structure and relational mapping. Removed the redundant MovieHall model for clarity and optimization.

frezworx committed:
Add serializers for Actor, Genre, and CinemaHall models

Introduced `ActorSerializer`, `GenreSerializer`, and `CinemaHallSerializer` to handle serialization and deserialization for their respective models. Added `create` and `update` methods to each serializer to support instance creation and updates. This enhances flexibility when working with these models in the API.

frezworx committed:
Add models for Actor, Genre, and CinemaHall, and relations to Movie

This migration introduces new models: Actor, Genre, and CinemaHall. Additionally, it establishes many-to-many relationships between Movie and both Actor and Genre. These changes set up the foundational structure for handling cinema-related entities and their interactions.

frezworx committed:
Refactor movie views to preparation for class-based views.

Replaced function-based views with a scaffold for class-based views using `APIView`. Existing functionality is commented out to enable a step-by-step migration. This change simplifies future expansion and improves code maintainability.

frezworx committed:
Add viewsets and routers for movies and cinema halls

Replaced function-based views with viewsets for better modularity and functionality. Updated URLs to use DRF routers and streamlined endpoint management. Introduced new API views for genres and actors using class-based views.

frezworx committed:
Add viewsets and routers for movies and cinema halls

Replaced function-based views with viewsets for better modularity and functionality. Updated URLs to use DRF routers and streamlined endpoint management. Introduced new API views for genres and actors using class-based views.

frezworx committed:
Refactor Genre and Actor views, update URL patterns.

Replaced `GenreAPIView` with `GenreList` and added `GenreDetail` for better REST structure. Simplified `ActorDetail` and `ActorList` by using generic views. Updated URL patterns and serializers to align with these changes.

mateacademy-ai-mentor commented on .gitignore:
```diff
@@ -1,9 +1,10 @@
 .vscode/
 *.iml
-.env
+.envпше
```
There seems to be a typo in the `.envпше` entry. It should likely be `.env` to correctly ignore environment variable files.

mateacademy-ai-mentor reviewed (approved):
****  
Great job on implementing the task requirements! 🎉 Your code is well-structured and meets most of the task requirements. The only minor issue is the typo in the `.gitignore` file entry, which should be corrected to `.env` to properly ignore environment files. Additionally, consider grouping imports using `()` to enhance code readability. Keep up the excellent work, and happy coding! 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

frezworx committed:
Add PUT, PATCH, and DELETE methods to GenreDetail view

Implemented functionality to update (PUT/PATCH) and delete (DELETE) genres in the `GenreDetail` view. Also fixed a typo in `.gitignore` for `.env` file exclusion.
